client/components/PuzzleItem/PuzzleItem.jsx

```diff
@@ -5,16 +5,17 @@ import x from "../../assets/x.png";
 import check from "../../assets/check.png";
 import "./PuzzleItem.css";
 
-export function PuzzleItem({ puzzle }) {
+export function PuzzleItem({ puzzle, puzzleCompletion }) {
   const [show, setShow] = useState(false);
   const handleClose = () => setShow(false);
   const handleShow = () => setShow(true);
 
   const { puzzle_id: puzzleID, title: puzzleTitle, description: puzzleDescription } = puzzle;
+  const isCompleted = puzzleCompletion ? puzzleCompletion.progress === 1 : false;
```

Comment:
not sure if this is necessary, since you can place `puzzleCompletion.progress === 1` in line 18 in place of `isCompleted`. or, even better yet, you can just do `puzzleCompletion.progress` (although i'm sure you have it set as such to handle the future case where a finished puzzle is not 1).

or are we checking if it's null here? either way, it feels kinda gross and less readable as a ternary followed by another ternary in line 18. you would probably get away with
```js
const isCompleted = puzzleCompletion?.progress === 1
```
since `undefined === 1` evaluates to false anyways. at that point you can just get rid of `isCompleted` in line 18 and switch it with the evaluation above

Reply:
ends up being more readable imho